model/src/main/kotlin/Defect.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+
+import java.net.URI
+import java.time.Instant
+
+/**
+ * A data class defining a model for software defects.
+ *
+ * Instances of this class are created by advisor implementations that retrieve information about known defects in
+ * packages.
+ */
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+data class Defect(
+    /**
+     * The (external) ID of this defect. This is a string used by a concrete issue tracker system to reference this
+     * defect, such as a bug ID or ticket number.
+     */
+    val id: String,
+
+    /**
+     * The URL pointing to the source of this defect. This is typically a reference into the issue tracker system that
+     * contains this defect.
+     */
+    val url: URI,
```

Comment:
IMHO, the URL should not be nullable, because - as mentioned before - it also allows tracking the source of the defect. A non-nullable URL is also in-line with the vulnerabilities model: here the URLs of references are mandatory, too.

Reply:
I see. I thought there may be cases where there is no URL or where one does not want to share it because with the outside world because its internal. Anyhow, let's leave it as is. Can be changed later if needed.